org.adempiere.ui/src/org/compiere/print/DrillReportCtl.java

```diff
@@ -222,30 +222,35 @@ public void initDrillTableMap() {
 				+ "FROM AD_Column c "
 				+ " INNER JOIN AD_Column used ON (c.ColumnName=used.ColumnName)"
 				+ " INNER JOIN AD_Table t ON (used.AD_Table_ID=t.AD_Table_ID AND t.IsView='N' AND t.AD_Table_ID <> c.AD_Table_ID AND t.IsShowInDrillOptions='Y')"
+				+ " INNER JOIN AD_Tab tab ON (t.AD_Table_ID = tab.AD_Table_ID AND tab.isActive = 'Y') "
+				+ " INNER JOIN AD_Window_Access w ON (tab.AD_Window_ID = w.AD_Window_ID AND w.isActive = 'Y') "
 				+ " INNER JOIN AD_Column cKey ON (t.AD_Table_ID=cKey.AD_Table_ID AND cKey.IsKey='Y')"
 				+ " INNER JOIN AD_Element e ON (cKey.ColumnName=e.ColumnName) "
-				+ "WHERE c.AD_Table_ID=? AND c.IsKey='Y' "
-				+ "ORDER BY 3 ";
+				+ "WHERE c.AD_Table_ID=? AND w.AD_Role_ID = ? AND c.IsKey='Y' "
+				+ "GROUP BY 1, 2, 3, 4 ORDER BY 3 ";
 			boolean trl = !Env.isBaseLanguage(Env.getCtx(), "AD_Element");
 			if (trl)
 				sql = "SELECT t.AD_Table_ID, t.TableName, et.PrintName, NULLIF(et.PO_PrintName,et.PrintName) "
 					+ "FROM AD_Column c"
 					+ " INNER JOIN AD_Column used ON (c.ColumnName=used.ColumnName)"
 					+ " INNER JOIN AD_Table t ON (used.AD_Table_ID=t.AD_Table_ID AND t.IsView='N' AND t.AD_Table_ID <> c.AD_Table_ID AND t.IsShowInDrillOptions='Y')"
+					+ " INNER JOIN AD_Tab tab ON (t.AD_Table_ID = tab.AD_Table_ID AND tab.isActive = 'Y') "
+					+ " INNER JOIN AD_Window_Access w ON (tab.AD_Window_ID = w.AD_Window_ID AND w.isActive = 'Y') "
 					+ " INNER JOIN AD_Column cKey ON (t.AD_Table_ID=cKey.AD_Table_ID AND cKey.IsKey='Y')"
 					+ " INNER JOIN AD_Element e ON (cKey.ColumnName=e.ColumnName)"
 					+ " INNER JOIN AD_Element_Trl et ON (e.AD_Element_ID=et.AD_Element_ID) "
-					+ "WHERE c.AD_Table_ID=? AND c.IsKey='Y'"
+					+ "WHERE c.AD_Table_ID=? AND w.AD_Role_ID = ? AND c.IsKey='Y' "
 					+ " AND et.AD_Language=? "
```

Comment:
this is incomplete, need to cater for included role.

Reply:
@hengsin 
Done!

Reply:
this doesn't looks correct, missing the close bracket.

Reply:
Fixed